components/eventing-controller/testing/subscriber.go

```diff
@@ -0,0 +1,103 @@
+package testing
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/avast/retry-go"
+	pkgerrors "github.com/pkg/errors"
+)
+
+type Subscriber struct {
+	addr   string
+	t      *testing.T
```

Comment:
Why the test suite reference is part of the subscriber? I can see the usage in `start()` and `shutdown()` can be replaced with returning an error to the caller, and the caller is free to fail the test.
The same can happen with the handler functions, we can return `4XX` or `5XX` and check the response instead of failing the test.

Is returning an error from the goroutine is the reason that you avoided returning an error? If yes, do we have a strong reason to start and shut down the server in goroutines?

Reply:
Right, removed t from Subscriber.